doc/source/container_builder.rst

```diff
@@ -0,0 +1,276 @@
+
+
+Installing and Setting up vagrant
+---------------------------------
+
+.. note:: Optional step
+
+    This step can be skipped if there are no complex provision tasks
+    of the building environment required.
+
+Installing vagrant is well documented at
+https://docs.vagrantup.com/v2/installation/index.html
+
+Access to a machine started by vagrant is done through ssh exclusively.
+Because of that an initial key setup is required in the box vagrant should
+start. The kiwi provided build boxes includes the public key of the vagrant
+key pair and thus allows access. It is important to understand that the
+private vagrant key is not a secure key because the private key is not
+protected. However this is not a problem because vagrant creates a new
+key pair for each machine it starts. In order to allow vagrant the initial
+access and the creation of a new key pair, it's required to provide access
+to the insecure vagrant private key. The following commands should not be
+executed as root, but rather as the user intending to build images.
+
+.. code:: bash
+
+    $ mkdir -p ~/.dice/key
+    $ cp -a /usr/share/doc/packages/ruby2.1-rubygem-dice/key ~/.dice/key
+
+
+Building with dice
+------------------
+
+If you build in a contained environment, there is no need to have KIWI
+installed on the host system. KIWI is part of the container and is only
+called there. However, a KIWI image description and some metadata
+defining how to run the container are required as input data.
+
+Selecting a KIWI Template
+-------------------------
+
+If you don't have a KIWI description select one from the templates
+provided at the GitHub project hosting example appliance descriptions.
+
+.. code:: bash
+
+    $ git clone https://github.com/SUSE/kiwi-descriptions
+
+The descriptions hosted here also provides a default :file:`Dicefile`
+as part of each image description.
+
+The Dicefile
+------------
+
+The Dicefile is the configuration file for the dice buildsystem backend.
+All it needs to know for a plain docker based build process is the
+selection of the buildhost to be a docker container. The Dicefile's
+found in the above mentioned appliance descriptions look all like the
+following:
+
+.. code:: ruby
+
+    Dice.configure do |config|
+      config.buildhost = :DOCKER
+    end
+
+Building with dice
+------------------
+
+If you have choosen to just use the default dice configuration as
+provided with the example appliance descriptions, the following example
+command will build the image.
+
+.. code:: bash
+
+    $ cd <git-clone-result-kiwi-descriptions>
+
+    $ dice build suse/x86_64/suse-leap-42.1-JeOS
+    $ dice status suse/x86_64/suse-leap-42.1-JeOS/
+
+
+Buildsystem backends
+--------------------
+
+dice currently supports three build system backends:
+
+1. `host buildsystem` - dice builds on the host like if you would call
+   kiwi on the host directly
+
+2. `vagrant buildsystem` - dice uses vagrant to run a virtual system which
```

Comment:
Depending on context, do you mean Vagrant (as a project?) or `command:`vagrant` as a command?